plugins/home/backstage-plugin-home-markdown/src/MarkdownCard/Content.tsx

```diff
@@ -14,34 +14,80 @@
  * limitations under the License.
  */
 
-import React, { useEffect, useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Alert } from '@material-ui/lab';
 import { Progress, MarkdownContent } from '@backstage/core-components';
-import { useGithubFile } from './useGithubFile';
 import {
   useApi,
   githubAuthApiRef,
   SessionState,
+  useApiHolder,
+  ApiHolder,
 } from '@backstage/core-plugin-api';
 import { MarkdownContentProps } from './types';
 import { Button, Grid, Typography, Tooltip } from '@material-ui/core';
+import useAsync from 'react-use/lib/useAsync';
+import { githubApiRef, GithubClient, GithubApi } from '../apis';
+
+const getGithubClient = (apiHolder: ApiHolder) => {
+  let githubClient: GithubApi | undefined = apiHolder.get(githubApiRef);
+  if (!githubClient) {
+    const auth = apiHolder.get(githubAuthApiRef);
+    if (auth) {
+      githubClient = new GithubClient({ githubAuthApi: auth });
+    }
+  }
+  if (!githubClient) {
+    throw new Error(`Failed to get the github client`);
```

Comment:
Probably good to add the name of the plugin to the error for traceability when we read these?

Reply:
good point.